tests/e2e/regression/lowering_config.mlir

```diff
@@ -31,8 +31,8 @@ func.func @lowering_config_test() {
     lowering_config = <tile_sizes = [[0, 7, 7, 64, 0, 0, 0], [6, 7, 1, 32, 0, 0, 0], [0, 0, 0, 0, 3, 1, 4], [0, 0, 0, 0, 0, 0, 0]]>,
     translation_info = <CPUConvTileAndDecomposeExpert>>
 func.func @conv() {
-  %input = util.unfoldable_constant dense<1.0> : tensor<36x7x7x512xf32>
-  %filter = util.unfoldable_constant dense<1.0> : tensor<3x3x512x512xf32>
+  %input = util.unfoldable_constant dense<1.0> : tensor<12x7x7x512xf32>
+  %filter = util.unfoldable_constant dense<1.0> : tensor<3x3x512x128xf32>
```

Comment:
This test case is taking 33 seconds on riscv (matching the findings on your machine).
https://github.com/openxla/iree/actions/runs/7953949162/job/21736330918?pr=16472#step:8:1114

That is too slow for a single unit test still. I still have little idea what this regression test is actually trying to provide coverage for, but I highly doubt it is worth taking more then 5 seconds of CI time...

Maybe @hanhanW or @MaheshRavishankar could weigh in.

Reply:
The e2e tests are for preset compilation_info. We probably want the test only running on x86 machines because those tile sizes are mostly from x86 setup. For riscv, they might need different tile sizes.

Reply:
Yeah, not sure it is worth testing on RISC-V. If you want to maybe just use a really really small convolution size.

Reply:
If the purpose is for a compilation benchmark, should we move it to "hostonly" region?
https://github.com/openxla/iree/blob/a567e70ff3d10fd0e6688b96c61680929442ab53/tests/e2e/regression/CMakeLists.txt#L13-L31

Reply:
@ScottTodd Do we always use x86 as host? If so, I think it makes sense to move the test to "hostonly".

Reply:
> @ScottTodd Do we always use x86 as host? If so, I think it makes sense to move the test to "hostonly".

No, we run some tests on arm, e.g. macos_arm64
* Sample logs https://github.com/openxla/iree/actions/runs/8190744586/job/22398393857
* CI job spec: https://github.com/openxla/iree/blob/0545746bddea74f1a9f0b8ef5adfe0c380770145/.github/workflows/ci.yml#L158

Reply:
I see that `compiler/` has `lit` tests for different targets.
e.g
`compiler/src/iree/compiler/Codegen/LLVMCPU/test/select_x86_64_lowering_strategy.mlir`
Can anyone share what's the purpose of `tests/e2e/regression/lowering_config.mlir`?
Is it test for compilation speed?

Reply:
The purpose is to allow tuning at op level. It verifies that the preset configuration is correct end-to-end. The configurations are picked up for x86 machines. So it would be good if we can mark this test is x86only.